### ISSUE TITLE
grasp: fix bugs in evaluation csv output

### ISSUE THREAD

**Issue by ahundt** [jhu-lcsr/costar_plan#419]
Right now the EvaluationCallback outputs data that is unclear, not well labeled, and difficult to process with a computer like the following:
```
metrics:
['loss', 'acc', 'mean_pred', 'mean_true']results:[0.7855481931074279, 0.4886413221095128, 0.4157902227250883, 0.6456840137114366]in epoch0
metrics:
['loss', 'acc', 'mean_pred', 'mean_true']results:[0.6683563708088858, 0.5940947341200123, 0.5337935280938244, 0.6456840137114366]in epoch1
metrics:
['loss', 'acc', 'mean_pred', 'mean_true']results:[0.6872151293578518, 0.5971330640766156, 0.529389139877771, 0.6456840137114366]in epoch2
metrics:
['loss', 'acc', 'mean_pred', 'mean_true']results:[0.7247951157001012, 0.5564973522563912, 0.48443339826308723, 0.6456840137114366]in epoch3
metrics:
['loss', 'acc', 'mean_pred', 'mean_true']results:[0.7650169895009372, 0.5084138371191986, 0.42087575651521847, 0.6456840137114366]in epoch4
metrics:
['loss', 'acc', 'mean_pred', 'mean_true']results:[0.6787171044290279, 0.6053443452666653, 0.5418462624019387, 0.6456840137114366]in epoch5
metrics:
['loss', 'acc', 'mean_pred', 'mean_true']results:[0.7562137324277087, 0.5448737931819433, 0.46043223574909653, 0.6456840137114366]in epoch6
metrics:
['loss', 'acc', 'mean_pred', 'mean_true']results:[0.7346843578259209, 0.5509037091299525, 0.45726398411289615, 0.6456840137114366]in epoch7
metrics:
['loss', 'acc', 'mean_pred', 'mean_true']results:[0.7563057614385329, 0.5187909016022897, 0.44024812667545, 0.6456840137114366]in epoch8
metrics:
['loss', 'acc', 'mean_pred', 'mean_true']results:[0.7090050374346661, 0.5710813346095139, 0.4862351903791144, 0.6456840137114366]in epoch9
metrics:
['loss', 'acc', 'mean_pred', 'mean_true']results:[0.7077712849208876, 0.578092864695118, 0.48371206177064807, 0.6456840137114366]in epoch10
metrics:
['loss', 'acc', 'mean_pred', 'mean_true']results:[0.6881675539899881, 0.5796977262833699, 0.5178339336045381, 0.6456840137114366]in epoch11
metrics:
['loss', 'acc', 'mean_pred', 'mean_true']results:[0.7919484197658951, 0.5338267381466988, 0.4353226039784442, 0.6456840137114366]in epoch12
metrics:
['loss', 'acc', 'mean_pred', 'mean_true']results:[0.695407231715103, 0.5833125595037032, 0.5211992866104836, 0.6456840137114366]in epoch13
metrics:
['loss', 'acc', 'mean_pred', 'mean_true']results:[0.7570976986162784, 0.5543159870781996, 0.46773160664244545, 0.6456840137114366]in epoch14
metrics:
['loss', 'acc', 'mean_pred', 'mean_true']results:[0.7027630792262051, 0.5588345291152778, 0.5027538028256326, 0.6456840137114366]in epoch15
metrics:
['loss', 'acc', 'mean_pred', 'mean_true']results:[0.7008478321516977, 0.5529448434074336, 0.4902894058583746, 0.6456840137114366]in epoch16
```

It should be outputting data in csv format like this, but contain clear labels for both training and evaluation results:
```
epoch,acc,loss,mean_pred,mean_true
0,0.6816131977159609,0.591385887340309,0.3911735322043377,0.39109880442099504
1,0.6854876960150866,0.5866959395977954,0.3927573588137471,0.39280068702914905
2,0.6882515489858183,0.5824849765246743,0.39131029933170647,0.39126857712112434
3,0.6905108323914085,0.5792971351592257,0.3922179092827581,0.3922527842747541
4,0.6914232126802361,0.5769776624376451,0.39102937375644065,0.3910127308625779
5,0.6933334524196069,0.5740537829756905,0.3916296379332162,0.39165383063717346
6,0.6950121840310018,0.5721602697173286,0.39193637086401123,0.39192748536417765
7,0.6959275323301624,0.5706723322728829,0.3909772732450243,0.3909480273864246
8,0.6969562605002287,0.5688451061838357,0.39278949857489465,0.39278287889049995
9,0.6981535735474605,0.5673496792034025,0.3918753704100974,0.39191620684474526
10,0.6992683747091514,0.5655966855771928,0.3914568176889487,0.39147634112213286
11,0.6995135359237531,0.5649262269947426,0.3912597288158267,0.39126204753988136
12,0.7013454196249443,0.5631527170166197,0.39104900448994007,0.39102341553874037
13,0.7021218625567555,0.5619603466936979,0.3925666820361627,0.3926178548922793
14,0.7026258380275787,0.5610938659418636,0.391251027434994,0.3912282117446001
15,0.7031482159163154,0.560136832961388,0.3920124145795175,0.3919975315430301
16,0.7041953458376458,0.5585959092155275,0.39123200137078507,0.39127213881017464
```

Here is the TODO list:

- [x] output in csv format
- [x] give different but clear labels to training loss and metrics, as well as validation loss and metrics
    - Here is an **example** of how labels can be, but it should support changes in these variables and their names:
        - epoch
        - train_acc
        - train_loss
        - train_mean_pred
        - train_mean_true
        - eval_acc
        - eval_loss
        - ...
